integreat_compass/cms/models/users/user.py

```diff
@@ -0,0 +1,135 @@
+        """
+        email = self.normalize_email(email)
+        user = self.model(email=email, display_name=display_name, **extra_fields)
+
+        if group and group in roles.CHOICES:
+            user.groups.add(Group.objects.get(name=group))
+
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, display_name, password=None, **extra_fields):
+        """
+        Create a new super user
+
+        :param email: email address of the user, also used in lieu of username
+        :type email: str
+
+        :param display_name: name used for comments etc.
+        :type display_name: str
+
+        :param password: user password
+        :type password: str
+
+        :param extra_fields: additional fields
+        :type extra_fields: dict
+
+        :return: the newly created user
+        :rtype: ~integreat_compass.cms.models.users.user.User
+        """
+        extra_fields.setdefault("is_staff", True)
+        extra_fields.setdefault("is_superuser", True)
+        return self.create_user(email, display_name, password, **extra_fields)
+
+
+class User(AbstractBaseUser, PermissionsMixin, AbstractBaseModel):
+    """
+    A custom User model that replaces the default Django User model.
+    """
+
+    email = models.EmailField(
+        unique=True,
+        verbose_name=_("email"),
+        help_text=_("Valid email address for this user"),
+    )
+    display_name = models.CharField(
+        max_length=128,
+        verbose_name=_("display name"),
+        help_text=_("This name will be shown as the author of votes, comments, etc."),
+    )
+    is_active = models.BooleanField(default=True)
+    is_staff = models.BooleanField(default=False)
+
+    objects = CustomUserManager()
+
+    USERNAME_FIELD = "email"
+    REQUIRED_FIELDS = ["display_name"]
+
+    @cached_property
+    def role(self):
+        """
+        We refer to Django user groups as roles.
+
+        :return: The role of this user
+        :rtype: ~integreat_compass.cms.models.users.role.Role
+        """
+        return self.groups.first().name if self.groups.exists() else None
+
+    def __str__(self):
+        """
+        This overwrites the default Django :meth:`~django.db.models.Model.__str__` method which would return ``User object (id)``.
+        It is used in the Django admin backend and as label for ModelChoiceFields.
+
+        :return: A readable string representation of the user
+        :rtype: str
+        """
+        return f"{self.display_name} ({self.email})"
+
+    def get_repr(self):
+        """
+        This overwrites the default Django ``__repr__()`` method which would return ``<User: User object (id)>``.
+        It is used for logging.
+
+        :return: The canonical string representation of the user
+        :rtype: str
+        """
+        return f"<User (id: {self.id}, display_name: {self.display_name}, role: {self.role}, is_staff: {self.is_staff})>"
```

Comment:
```suggestion
        fields = [
            f"id: {self.id}",
            f"email: {self.email}",
            f"display_name: {self.display_name}",
            f"group: {self.groups.first()}",
            f"is_staff: {self.is_staff}",
        ]
        return f"<User ({', '.join(fields)})>"
```
We could show the email as well here